ports/esp32/mpconfigport.h

```diff
@@ -259,6 +259,24 @@ typedef long mp_off_t;
 // board specifics
 #define MICROPY_PY_SYS_PLATFORM "esp32"
 
+// Enable native USB peripheral CDC via TinyUSB
+#ifndef MICROPY_HW_USB_CDC
+#define MICROPY_HW_USB_CDC (CONFIG_USB_OTG_SUPPORTED)
+#endif
+
+// Enable serial over USB/JTAG serial peripheral
+#ifndef MICROPY_HW_ESP_USB_SERIAL_JTAG
+#define MICROPY_HW_ESP_USB_SERIAL_JTAG (!MICROPY_HW_USB_CDC && \
+    (CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED || (        \
+    ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 1, 0) && ( \
+    CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG || \
+    CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG))))
```

Comment:
This is pretty ugly but after we drop 5.0.x support it can be simplified to `(!MICROPY_HW_USB_CDC && CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED)`